src/or/main.c

```diff
@@ -1960,6 +1958,29 @@ check_authority_cert_callback(time_t now, const or_options_t *options)
   return CHECK_V3_CERTIFICATE_INTERVAL;
 }
 
+/**
+ * Scheduled callback: Run directory-authority voting functionality.
+ *
+ * The schedule is a bit complicated here, so dirvote_act() manages the
+ * schedule itself.
+ **/
+static int
+dirvote_callback(time_t now, const or_options_t *options)
+{
+  time_t next = TIME_MAX;
+  if (authdir_mode_v3(options)) {
+    next = dirvote_act(options, now);
+  }
+  if (BUG(next == TIME_MAX)) {
+    return 3600;
+  }
+  if (BUG(next <= now)) {
+    return 1;
```

Comment:
So, I think this is impossible for the same reason it couldn't happen in #25948 -- the "next" value here is based on the input time to dirvote_act(), so unless dirvote_act() is buggy, it can't be earlier than the input.

Reply:
Indeed.